cc/dual_net/inference_cache.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cc/dual_net/inference_cache.h"
+
+namespace minigo {
+
+InferenceCache::CompressedFeatures InferenceCache::CompressFeatures(
+    const DualNet::BoardFeatures& features) {
```

Comment:
I don't remember if LZ ever implemented this but features (19x19x17) >> zobrist hash (64)

Reply:
I originally thought about using the Zobrist hash for the cache key but wasn't happy about effectively ignoring 14 of the feature planes on a cache hit.

However, your comment did give the idea that we can instead use concatenate the most recent 8 Zobrist hashes and 1 bytes for the player. This makes the cache key only 65 bytes instead of 768. I'll add that in a followup PR.

Thanks! 10x impact :)